packages/template-retail-react-app/app/hooks/use-category-bulk.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useCommerceApi, useAccessToken} from 'commerce-sdk-react-preview'
+import {useQueries} from '@tanstack/react-query'
+import {getConfig} from 'pwa-kit-runtime/utils/ssr-config'
+import useMultiSite from './use-multi-site'
+
+/**
+ * a hook that parallelly and individually fetches category based on the given ids
+ * @param ids - list of categories ids to fetch
+ * @param queryOptions -  react query options
+ * @return list of react query results
+ */
+export const useCategoryBulk = (ids, queryOptions) => {
```

Comment:
having `queryOptions` allows users to get access to query callbacks. For example
```
const res = useCategoryBulk(ids, {onSuccess: (data) => {
  // do something with each query result
}})
```